### PR TITLE
Update species removal confirmation

### PR DESCRIPTION
**Pull request by ridwanamo**
## Description
Use shared component DeletionConfirmation to show the removing message confirmation. By using the shared component, we didnt need the SpeciesRemove.tsx component anymore and move any other code to the parent component (SpeciesTitleArea.tsx).

The tricky was to do the styling for the different screen size (see XD in jira ticket)

**Check that the google analytics is still working.**

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1733

## Deployment URL(s)
http://species-removal-confirmation.review.ensembl.org

## Views affected
species home page (go to any species home page and click on the remove species button)

### CODE REVIEW

**Comment by azangru** [comment]
Why the vertical jump here?

https://user-images.githubusercontent.com/6834224/193828900-2d387e83-5a76-491a-b585-c5984f719228.mov

**Comment by azangru** [comment]
Broken button

https://user-images.githubusercontent.com/6834224/193846637-d384db87-a03f-4e18-b77c-98014d54d848.mov

**Comment by azangru** [comment]
Using media queries can get you in a situation when there is plenty of space in the container, but we aren't using it:

https://user-images.githubusercontent.com/6834224/193847524-93d84e2e-93b5-4807-8d74-318d7de6c72c.mov

Ideally, we would solve this with CSS container queries, but they are still very new and not supported in Firefox; but what we could do instead is use the `useResizeObserver` hook to update the class name on the container. That would allow us to only change the layout when there is really no available space

File: src/content/app/species/components/species-title-area/SpeciesTitleArea.scss [hunk 1]
```diff
@@ -2,61 +2,17 @@
...
-}
-
 .speciesTitleArea {
   display: grid;
-  grid-template-areas: 'icon species-name usage-toggle search remove';
+  grid-template-areas:
+    'icon species-name usage-toggle search remove'
+    'remove-message . . . .';
   grid-template-columns: 60px fit-content(60%) minmax(200px, 680px) auto 1fr;
+  grid-template-rows: 80px auto auto auto auto;
```
Comment (azangru) [inline src/content/app/species/components/species-title-area/SpeciesTitleArea.scss:11]: why so many rows?

**Comment by azangru** [comment]
Is the red text allowed to get under the species icon?

https://user-images.githubusercontent.com/6834224/193853296-77dc36b6-9f38-479a-be92-4b7172d7d00d.mov

File: src/content/app/species/components/species-title-area/SpeciesTitleArea.scss [hunk 2]
```diff
@@ -111,3 +79,65 @@ $container-padding-right: 20px;
...
+
+@media screen and (max-width: 1040px) {
+  .speciesTitleArea {
+    grid-template-areas:
+      'icon species-name .'
+      '. usage-toggle remove'
+      '. remove-message .'
+      '. search .';
+    grid-template-columns: 60px 150px 1fr;
+    grid-template-rows: 80px 0px auto 40px;
```
Comment (azangru) [inline src/content/app/species/components/species-title-area/SpeciesTitleArea.scss:134]: Why do you need a 0-pixel-high and a 40-pixel-high row?

**Comment by ens-ma7** [comment]
> Is the red text allowed to get under the species icon?
> 
>  Screen.Recording.2022-10-04.at.15.54.32.mov

This should be fixed with my latest commit

**Comment by ens-ma7** [comment]
> Broken button

Fixed

File: src/content/app/species/components/species-title-area/SpeciesTitleArea.tsx [hunk 3]
```diff
@@ -14,27 +14,36 @@
...
-import { fetchPopularSpecies } from 'src/content/app/species-selector/state/speciesSelectorSlice';
+import {
+  fetchPopularSpecies,
+  deleteSpeciesAndSave
+} from 'src/content/app/species-selector/state/speciesSelectorSlice';
 
 import SpeciesUsageToggle from './species-usage-toggle/SpeciesUsageToggle';
-import SpeciesRemove from './species-remove/SpeciesRemove';
 
 import { RootState } from 'src/store';
```
Comment (imransl) [inline src/content/app/species/components/species-title-area/SpeciesTitleArea.tsx:48]: The import groupings are not right in this file

File: src/content/app/species/components/species-title-area/SpeciesTitleArea.tsx [hunk 4]
```diff
@@ -14,32 +14,51 @@
...
+import { useNavigate } from 'react-router-dom';
 
 import { useAppSelector, useAppDispatch } from 'src/store';
 
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import useSpeciesAnalytics from 'src/content/app/species/hooks/useSpeciesAnalytics';
+import useResizeObserver from 'src/shared/hooks/useResizeObserver';
+
 import { getDisplayName } from 'src/shared/components/selected-species/selectedSpeciesHelpers';
```
Comment (imransl) [inline src/content/app/species/components/species-title-area/SpeciesTitleArea.tsx:28]: The import groupings are okay now except for this one

**Review by imransl — APPROVED** [review]

**Review by azangru — APPROVED** [review]

**Pull request merged**